mlir-sycl/doc/design/SYCLGridRepresentation.md

```diff
@@ -11,57 +11,48 @@ pointer types, `llvm.struct` for structured types and `sycl.cast` is the
 operation used to perform a casting from a SYCL type to its base class (or one
 of its base classes in case of multiple inheritance).
 
-## The `sycl.call` operation
-
-The SYCL dialect includes a `sycl.call` operation representing function
-calls. As these SYCL index space identifiers are obtained through member
-function calls, e.g., `sycl::nd_item::get_global_id()`, `sycl.call` is the first
-candidate that comes to mind for this task. However, this operation is too
-low-level to do the work for us. If we look at this example:
-
-```MLIR
-%0 = sycl.call() {FunctionName = @get_global_id, MangledFunctionName = @..., TypeName = @nd_item} : (!memref<?xsycl_nd_item_1_>) -> !sycl_id_1_
-%1 = sycl.call() {FunctionName = @get_local_id, MangledFunctionName = @..., TypeName = @nd_item} : (!memref<?xsycl_nd_item_1_>) -> !sycl_id_1_
-```
-
-We can see how, even though `%0` and `%1` have a very different nature, they
-have the same type and defining operation, in fact, they only differ on the
-`FunctionName` (and `MangledFunctionName`) attribute of their defining
-operations. As the target is to make memory access analysis easier, we propose
-index space identifiers of different nature should differ at least on their
-defining operation.
-
-Also, in order to perform memory access analysis, these operations should be
-presented in a canonical form. A canonicalization pattern checking every
-`sycl.call` operation in the MLIR module and replacing it with different
-`sycl.call` operations can be a bit cumbersome, thus we propose these operations
-to have different defining operations.
+## SYCL ND-range operations
+
+These operations query ND-range components such as the number of work items or
+the global id. These can be ND `id` or `range` instances or `i32` scalars,
+depending on the operation.
+
+See the SYCL dialect documentation for more information on each operation.
+
+### Lowering
+
+The ND-range operations lower to platform-specific built-ins, e.g., SPIR-V
+built-ins and the necessary casts. If the array subscript order differs, this
+should be taken into account.
```

Comment:
Is the subscript order already handled, or is this a TODO?

Reply:
Taken into account already. Made it explicit now